src/data/index.ts

```diff
@@ -0,0 +1,25 @@
+import { getCurrentSemester } from './api';
+
+export interface Fraction {
+	numerator: number;
+	denominator: number;
+}
+
+export interface CourseItem {
+	name: string;
+	weight: number;
+	grade?: number | Fraction;
+}
+
+export interface Course {
+	name: string;
+	crn?: number;
+	items: CourseItem[];
+}
+
+export interface Semester {
+	name: string;
+	courses: Course[];
+}
+
+export { getCurrentSemester };
```

Comment:
is importing `getCurrentSemester` and then exporting it a good idea or would it be better to be more explicit where we need to use it and `import { getCurrentSemester } from './api';` in those files rather than here.

Do we have a reason for this?

Reply:
I feel like I have poorly named the `data` folder, and really it itself should probably be named `api`. Re exporting would allow us to import `src/api` and attempt to only expose things that are intended to be accessible outside that module (for example, we don't want people importing the mock data `data/api/mock`. There is no way to block it but this may make it easier to catch).